src/main/java/in/mohamedhalith/dao/EmployeeDAO.java

```diff
@@ -116,20 +116,47 @@ public Employee getEmployee(String username) throws DBException, ValidationExcep
 	 * @param employee
 	 * @param type
 	 * @param duration
+	 * @return
 	 * @throws DBException
+	 * @throws ValidationException
 	 */
-	public void updateLeaveBalance(Employee employee, String type, int duration) throws DBException {
+	public boolean updateLeaveBalance(Employee employee, String type, int duration)
+			throws DBException, ValidationException {
 		Connection connection = null;
 		PreparedStatement statement = null;
 		String username = employee.getUsername();
-		type = type.toLowerCase();
+		String query = null;
+		int leave = -1;
+		boolean isUpdated = false;
 		try {
 			connection = ConnectionUtil.getConnection();
-			String query = "update employees set " + type + " = " + type + "- ? where username = ?";
+			switch (type) {
+			case "SickLeave":
+				int sickLeave = employee.getSickLeave();
+				leave = sickLeave - duration;
+				query = "update employees set sickleave = ? where username = ?";
+				break;
+			case "CasualLeave":
+				int casualLeave = employee.getCasualLeave();
+				leave = casualLeave - duration;
+				query = "update employees set casualleave = ? where username = ?";
+				break;
+			case "EarnedLeave":
+				int earnedLeave = employee.getEarnedLeave();
+				leave = earnedLeave - duration;
+				query = "update employees set earnedleave = ? where username = ?";
+				break;
+			default:
+				throw new ValidationException("Invalid Leave type");
```

Comment:
why validation exception comes under D.A.O. change the exception name some other. After validation only you call D.A.O